### PR TITLE
GO-1927 Experience import request

### PR DESCRIPTION
**Pull request by KirillSto**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description

This PR introduces new RPC request that allows to download experience (aka usecase) from remote via URL

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents

https://linear.app/anytype/issue/GO-1927/implement-experience-downloading-function

### Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

File: util/builtinobjects/builtinobjects.go [hunk 1]
```diff
@@ -176,6 +178,50 @@ func (b *builtinObjects) CreateObjectsForUseCase(
 	return pb.RpcObjectImportUseCaseResponseError_NULL, nil
 }
 
+func (b *builtinObjects) CreateObjectsForExperience(ctx context.Context, spaceID, source string, isLocal bool) (err error) {
+	if isLocal {
+		return b.importArchive(ctx, spaceID, source)
+	}
+	resp, err := http.Get(source)
```
Comment (deff7) [inline util/builtinobjects/builtinobjects.go:185]: `//nolint: gosec`

File: util/builtinobjects/builtinobjects.go [hunk 2]
```diff
@@ -185,6 +231,11 @@ func (b *builtinObjects) inject(ctx context.Context, spaceID string, useCase pb.
 	if err = os.WriteFile(path, archive, 0644); err != nil {
 		return fmt.Errorf("failed to save use case archive to temporary file: %s", err.Error())
 	}
+	defer func() {
+		if err = os.Remove(path); err != nil {
+			log.Errorf("failed to remove temporary file %s: %s", path, err.Error())
```
Comment (AnastasiaShemyakinskaya) [inline util/builtinobjects/builtinobjects.go:236]: I think, it's not correct to log path

File: util/builtinobjects/builtinobjects.go [hunk 3]
```diff
@@ -176,6 +178,50 @@ func (b *builtinObjects) CreateObjectsForUseCase(
...
+		return fmt.Errorf("failed to fetch experience from '%s': not OK status code: %s", source, resp.Status)
+	}
+	defer func() {
+		if err = resp.Body.Close(); err != nil {
+			log.Errorf("failed to close response bode while downloading experience from '%s': %v", source, err)
+		}
+	}()
+
+	path := filepath.Join(b.tempDirService.TempDir(), time.Now().Format("tmp.20060102.150405.99")+".zip")
+	out, err := os.Create(path)
```
Comment (AnastasiaShemyakinskaya) [inline util/builtinobjects/builtinobjects.go:199]: This error can contain user local path. I think you can use function TransformError from utils os package to hide path.

File: util/builtinobjects/builtinobjects.go [hunk 4]
```diff
@@ -176,6 +178,50 @@ func (b *builtinObjects) CreateObjectsForUseCase(
...
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err = out.Close(); err != nil {
+			log.Errorf("failed to close temporary file while downloading experience from '%s': %v", source, err)
+		}
+		if err = os.Remove(path); err != nil {
+			log.Errorf("failed to remove temporary file: %v", err)
```
Comment (AnastasiaShemyakinskaya) [inline util/builtinobjects/builtinobjects.go:208]: The same

File: util/builtinobjects/builtinobjects.go (hunk 1 above)
Comment (KirillSto) [reply util/builtinobjects/builtinobjects.go:187]: Added nolint to http.Get

File: util/builtinobjects/builtinobjects.go (hunk 2 above)
Comment (KirillSto) [reply util/builtinobjects/builtinobjects.go]: Fixed

File: util/builtinobjects/builtinobjects.go (hunk 3 above)
Comment (KirillSto) [reply util/builtinobjects/builtinobjects.go:201]: Thank you for an advice of useful func! Fixed

File: util/builtinobjects/builtinobjects.go (hunk 4 above)
Comment (KirillSto) [reply util/builtinobjects/builtinobjects.go]: Done

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

**Pull request merged**